bin/elixir

```diff
@@ -159,58 +168,57 @@ while [ $I -le $LENGTH ]; do
     --pipe-to)
         S=3
         RUN_ERL_PIPE="$2"
-        case "$RUN_ERL_PIPE" in
-          -*) echo "--pipe-to : PIPEDIR cannot be a switch" >&2 && exit 1;;
-          *);;
-        esac
-
         RUN_ERL_LOG="$3"
-        case "$RUN_ERL_LOG" in
-          -*) echo "--pipe-to : LOGDIR cannot be a switch" >&2 && exit 1;;
-          *);;
-        esac
+        if [ "$(starts_with "$RUN_ERL_PIPE" "-")" ]; then
+          echo "--pipe-to : PIPEDIR cannot be a switch" >&2 && exit 1
+        elif [ "$(starts_with "$RUN_ERL_LOG" "-")" ]; then
+          echo "--pipe-to : LOGDIR cannot be a switch" >&2 && exit 1
+        fi
         ;;
     --werl)
         if [ "$OS" = "Windows_NT" ]; then ERL_EXEC="werl"; fi
         ;;
     *)
         while [ $I -le $LENGTH ]; do
-          I=$(($I + 1))
+          I=$((I + 1))
           set -- "$@" "$1"
           shift
         done
         break
         ;;
   esac
 
-  I=$(($I + $S))
+  I=$((I + S))
   shift $S
 done
 
-I=$(($E - 1))
+I=$((E - 1))
 while [ $I -ge 0 ]; do
   eval "VAL=\$E$I"
   set -- "$VAL" "$@"
-  I=$(($I - 1))
+  I=$((I - 1))
 done
 
 SELF=$(readlink_f "$0")
 SCRIPT_PATH=$(dirname "$SELF")
 
-if [ "$OSTYPE" = "cygwin" ]; then SCRIPT_PATH=$(cygpath -m "$SCRIPT_PATH"); fi
+# shellcheck disable=SC2039,SC2086
+if [ -n "$OSTYPE" ] && [ "$OSTYPE" = "cygwin" ]; then SCRIPT_PATH=$(cygpath -m "$SCRIPT_PATH"); fi
```

Comment:
Let's keep this one as it was then, as the `-n` is not buying us anything. :)

Reply:
done and ready to be merged